build.gradle

```diff
@@ -36,6 +47,51 @@ dependencies {
   mlcp files("mlcp/lib")
 }
 
+/**
+ * START: Disconnected gradle tasks
+ * Reference: https://github.com/derms/ml-gradle/blob/dev/examples/disconnected-project-using-plugins-and-gradlew/build.gradle
+ */
+gradle.taskGraph.whenReady { graph ->
+  if (graph.hasTask(downloadToProjectMavenRepo)) {
+    println project.gradle.gradleUserHomeDir
+    if (!project.gradle.gradleUserHomeDir.equals(new File(rootDir, projectGradleHome))) {
+      throw new GradleException("Please set the gradle user home property to $projectGradleHome on the gradle command line - e.g. \n " +
+              (System.getProperty("os.name").startsWith("Windows") ? "" : "./") +
+              "gradlew -Dgradle.user.home=$projectGradleHome <task_to_execute>")
+    }
+  }
+}
+
+task downloadToProjectMavenRepo(type: Copy) {
+  /*
+   * Include any configuration dependencies here that you want to copy the dependencies for.
+   * These are defined in the 'dependencies' block. E.g. you need to include
+   * configurations.compile.files if you want your java 'compile' dependencies downloaded
+   */
+  //configurations.compile.files //includes 'java' compile dependencies. Remove/add as needed
+  configurations.mlcp.files //includes 'mlcp' dependencies. Remove if not needed
+
+  from new File(gradle.gradleUserHomeDir, 'caches/modules-2/files-2.1') // correct as of gradle 4.7
+  into new File(rootDir, projectMavenRepo)
+  eachFile {
+    List<String> parts = it.path.split('/')
+    it.path = (parts[0].replace('.', '/') + '/' + parts[1]) + '/' + parts[2] + '/' + parts[4]
+  }
+  includeEmptyDirs false
+}
+
+def projectZipFilename = projectBaseZipFilename + '-' + new Date().format('yyyyMMddHHmmss') + '.zip'
```

Comment:
I inserted an extra `project.name + '-' +`, as there might be multiple gradle projects you'd like/need to share..